test/node/extended_json.test.ts

```diff
@@ -784,4 +791,12 @@ Converting circular structure to EJSON:
       expect(parsedUUID).to.deep.equal(expectedResult);
     });
   });
+
+  it('should only enumerate own property keys from input objects', () => {
+    const input = { a: 1 };
+    Object.setPrototypeOf(input, { b: 2 });
+    const string = EJSON.stringify(input);
+    expect(string).to.include(`"a":`);
+    expect(string).to.not.include(`"b":`);
```

Comment:
same comment here as in serializer.test.ts

Reply:
This is a little different, I can hardcode the JSON as is done in other tests here but the issue I've seen crop up a number of times when working on EJSON is the spacing changes which doesn't actually mean the format really changed. Currently this test isn't space sensitive, ideally I'd like us to add a JSON equal checker that is better at being agnostic about that stuff, we'll likely need one when we fix the EJSON corpus issues.

Reply:
So stringified ejson is valid json right?  The concern is circularly testing our bson library by using ejson.parse to test ejson.stringify and vice-versa.  But could we use JSON.parse?

```typescript
const input = { a: 1 };
Object.setPrototypeOf(input, { b: 2 });
const string = EJSON.stringify(input);
const parsed = JSON.pase(string)
expect(parsed).to.deep.equal({ a: 1 });
```

Reply:
Yea for this test that's fine, I'll update. but generally JSON.parse is lossy in certain scenarios that do matter for EJSON/BSON correctness, so it's not always viable

Reply:
interesting, do you have an example?

Reply:
I kept the assertion that checks the b string doesn't exist, since that's what we really care about seems worth keeping, and of course json parse + deep equal confirms it

Reply:
Numbers that exceed the precision supported by javascript